pipelines/short-read-alignment-vector/ShortReadAlignment.wdl

```diff
@@ -0,0 +1,467 @@
+    File indel_realigner_interval_list_file = RealignerTargetCreator.output_interval_list_file
+    File output_bam = IndelRealigner.output_file
+    File validate_samfile_report_file = ValidateSamFile.report_file
+    File samtools_stats_report_file = SamtoolsStats.report_file
+    File callable_loci_summary_file = GatkCallableLoci.summary_file
+  }
+}
+
+task ReadAlignment {
+  input {
+    String read_group_id
+    File fastq1
+    File fastq2
+    String sample_id
+    String output_sam_basename
+
+    ReferenceSequence reference
+    RunTimeSettings runTimeSettings
+  }
+
+  Float fastq_size = size(fastq1, "GiB") + size(fastq2, "GiB")
+  Float ref_size = size(reference.ref_fasta, "GiB") + size(reference.ref_fasta_index, "GiB") + size(reference.ref_dict, "GiB")
+  Float bwa_ref_size = ref_size + size(reference.ref_amb, "GiB") + size(reference.ref_ann, "GiB") + size(reference.ref_bwt, "GiB") + size(reference.ref_pac, "GiB") + size(reference.ref_sa, "GiB")
+  Float disk_multiplier = 2.5
+  Int disk_size = ceil(fastq_size + bwa_ref_size + (disk_multiplier * fastq_size) + 20)
+
+  command <<<
+    set -o pipefail
+    set -e
+
+    # TODO - need to set the read group id correctly here.
+    /bwa/bwa mem -M -T 0 -R '@RG\tID:~{read_group_id}\tSM:~{sample_id}' ~{reference.ref_fasta} ~{fastq1} ~{fastq2} > ~{output_sam_basename}.sam
+  >>>
+  runtime {
+    docker: runTimeSettings.bwa_docker
+    preemptible: runTimeSettings.preemptible_tries
+    memory: "3.75 GiB"
+    cpu: "1"
+    disks: "local-disk " + disk_size + " HDD"
+  }
+  output {
+    File output_sam = "~{output_sam_basename}.sam"
+  }
+}
+
+task ReadAlignmentPostProcessing {
+  input {
+    File input_sam
+    String output_bam_basename
+    ReferenceSequence reference
+    RunTimeSettings runTimeSettings
+  }
+
+  Int disk_size = (ceil(size(input_sam, "GiB")) * 3) + 20
+
+  command <<<
+
+    set -e
+    set -o pipefail
+
+    /bin/samtools view -bu ~{input_sam} |
+    /bin/samtools sort -n - |
+    /bin/samtools fixmate - - |
+    /bin/samtools sort - |
+    /bin/samtools calmd -b - ~{reference.ref_fasta} > ~{output_bam_basename}.bam
+
+  >>>
+
+  runtime {
+    docker: runTimeSettings.samtools_docker
+    preemptible: runTimeSettings.preemptible_tries
+    memory: "14 GiB"
+    cpu: "4"
+    disks: "local-disk " + disk_size + " HDD"
+  }
+  output {
+    File output_bam = "~{output_bam_basename}.bam"
+  }
+}
+
+task MergeSamFiles {
+  input {
+    Array[File] input_files
+    String output_filename
+    RunTimeSettings runTimeSettings
+  }
+
+  Int disk_size = (ceil(size(input_files, "GiB")) * 3) + 20
+
+  command {
+    java -Xms3500m -jar /root/picard.jar \
+      MergeSamFiles \
+      INPUT=~{sep=' INPUT=' input_files} \
+      OUTPUT=~{output_filename}
+  }
+  runtime {
+    docker: runTimeSettings.picard_docker
+    preemptible: runTimeSettings.preemptible_tries
+    disks: "local-disk " + disk_size + " HDD"
+    memory: "3.75 GiB"
```

Comment:
A note on the memory allocation strategy.  Google cloud VMs come in standard sizes (https://cloud.google.com/compute/all-pricing#n1_standard_machine_types).  The cheapest is a 1 cpu model with 3.75 G of memory (the next is 7.G w/2 cores...).  So I normally just ask for that 3.75 and in the Java invocation use -Xms to preallocate all the memory to the process.  This won't work for running on LSF.